x-pack/legacy/plugins/uptime/public/components/functional/charts/donut_chart_legend_row.tsx

```diff
@@ -33,6 +33,11 @@ export const DonutChartLegendRow = ({ color, content, message }: Props) => (
     <EuiFlexItemReducedMargin component="span" grow={false}>
       {message}
     </EuiFlexItemReducedMargin>
-    <EuiFlexItemAlignRight component="span">{content}</EuiFlexItemAlignRight>
+    <EuiFlexItemAlignRight
+      component="span"
+      data-test-subj={`xpack.uptime.donutChart.${message.toLowerCase()}`}
```

Comment:
Done in https://github.com/elastic/kibana/pull/52433/commits/5d789d024ab8ace0b1bd406a3c47f9c59e5a2382#diff-d51e4dd872235464063c1c97819da10d

Reply:
Done in 5d789d0

Reply:
I think I did a poor job explaining what I was looking for. Here're some modified snippets of what I had in mind, LMK what you think of this:

#### Chart Legend Row
This is where we need the attribute applied so the test utils can find our values:
```typescript
interface Props {
  'data-test-subj': string;
  ...
}

export const DonutChartRow = ({
  'data-test-subj': dts,
  ...
}) => (
  <EuiFlexGroup ...>
    <EuiFlexItemAlignRight component="span" data-test-subj={dts}>
      {content}
    </EuiFlexItemAlignRight>
  </EuiFlexGroup>
);
```

#### Chart Legend
Then in here we could just hardcode the props as part of our component definition, and we won't be relying on the translated message strings for our attributes.

```typescript
      <DonutChartLegendRow
        color={danger}
        content={down}
        message={i18n.translate('xpack.uptime.snapshot.donutChart.legend.downRowLabel', {
          defaultMessage: 'Down',
        })}
        data-test-subj="xpack.uptime.snapshot.donutChart.down"
      />
      <DonutChartLegendRow
        color={gray}
        content={up}
        message={i18n.translate('xpack.uptime.snapshot.donutChart.legend.upRowLabel', {
          defaultMessage: 'Up',
        })}
        data-test-subj="xpack.uptime.snapshot.donutChart.up"
      />
```

Doing it this way we will only be touching two files, and modifying the only components that should really care about these attributes; plus the full values are readable and searchable if someone is trying to trace these keys from the test files for some reason.

Reply:
Ah, I see. My thinking is that we may want to reuse this donut chart elsewhere. By making this a prefix rather than the full subject, and passing it in from the top we could re-use the donut chart elsewhere and give it a separate data-test-subj namespace.